app/api/recommend.py

```
@@ -110,30 +110,51 @@ def startup_event():
+            latency_ms = int((time.perf_counter() - start_time) *1000)
+            logger.info(f"[CACHE][HIT] ttl=$s key=%s", ttl, cache_key_masked, extra={
+                "traceId": trace_id,
+                "userId": user_id,
+                "endpoint": endpoint,
+                "status": 200,
+                "latencyMs": latency_ms
+            })
             return json.loads(cached)
         else: 
-            logger.info(f"[CACHE][MISS] key={cache_key_masked}")
+            logger.info(f"[CACHE][MISS] key=%s", cache_key_masked, extra={
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)